nextcord/client.py

```diff
@@ -811,6 +858,12 @@ async def start(self, token: str, *, reconnect: bool = True) -> None:
             An unexpected keyword argument was received.
         """
         await self.login(token)
+        if self._endpoint_on_start:
+            if client_secret is not None:
+                self.http.set_client_secret(client_secret)
```

Comment:
Should this warn if the client secret is none but the user wants the endpoint to be started, I think so